packages/fxa-settings/src/pages/Signin/SigninUnblock/container.tsx

```diff
@@ -79,6 +80,17 @@ const SigninUnblockContainer = ({
     authEmail: string = email,
     signInOptions?: SignInOptions
   ) => {
+    // If the query param hasLinkedAccount is defined, then we know the user
+    // came from email-first and was already prompted with the sync merge
+    // warning. The browser will automatically respond with { ok: true } without
+    // prompting the user if it matches the email the browser has data for.
+    const params = new URLSearchParams(location.search);
+    if (integration.isSync() && params.get('hasLinkedAccount') === undefined) {
```

Comment:
FWIW, I'm not sure if this is needed in `signinWithUnblockCode`? For the user to get to this page they would have already attempted a login and should have been prompted to merge account.

Reply:
Ohh, this is a good point. I saw some comment in content-server that made me think we needed it, but you're right that we navigate there after signin so users will have seen it on signin first.